src/common/spawn.c

```diff
@@ -25,11 +25,20 @@ reset_signals_and_limits(void)
 	signal(SIGPIPE, SIG_DFL);
 }
 
-static void
+bool
 set_cloexec(int fd)
 {
 	int flags = fcntl(fd, F_GETFD);
-	fcntl(fd, F_SETFD, flags | FD_CLOEXEC);
+	if (flags == -1) {
+		wlr_log(WLR_ERROR, "Unable to set the CLOEXEC flag: fnctl failed");
+		return false;
+	}
+	flags = flags | FD_CLOEXEC;
+	if (fcntl(fd, F_SETFD, flags) == -1) {
+		wlr_log(WLR_ERROR, "Unable to set the CLOEXEC flag: fnctl failed");
+		return false;
+	}
```

Comment:
We could use `wlr_log_errno()` instead, it just applies a `: strerr(errno)` to the end of the log message.

Reply:
Use of `wlr_log_errno()` implemented.